### PR TITLE
Support for Python 3.12

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        python_version: ['3.8', '3.9', '3.10', '3.11']
+        python_version: ['3.9', '3.10', '3.11', '3.12']
```
Comment (alexander-veit) [inline .github/workflows/main.yml]: Is there a reason, you removed 3.8 from the tests?

File: pyproject.toml [hunk 2]
```diff
@@ -21,9 +21,10 @@ packages = [
 
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
-dcicutils = "^8.0.0"
-tibanna-ff = ">=3.2.0"
+python = ">=3.8.1,<3.13"
+dcicutils = "^8.13.3"
+docutils = ">=0.10,<0.17,"
+tibanna-ff = "3.4.2.1b2"
```
Comment (alexander-veit) [inline pyproject.toml]: Adjust version upon release of tibanna-ff

File: pyproject.toml [hunk 3]
```diff
@@ -21,9 +21,10 @@ packages = [
 
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
-dcicutils = "^8.0.0"
-tibanna-ff = ">=3.2.0"
+python = ">=3.8.1,<3.13"
+dcicutils = "^8.13.3"
+docutils = ">=0.10,<0.17,"
```
Comment (alexander-veit) [inline pyproject.toml]: Why do we need this now?

**Review by alexander-veit — COMMENTED** [review]
See comments. `poetry.lock` needs to be regenerated

File: pyproject.toml [hunk 4]
```diff
@@ -21,9 +21,10 @@ packages = [
 
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
-dcicutils = "^8.0.0"
-tibanna-ff = ">=3.2.0"
+python = ">=3.8.1,<3.13"
```
Comment (alexander-veit) [inline pyproject.toml:24]: I would not exclude 3.8.0 here if there is no good reason. Technically that would be a breaking change.

Comment (dmichaels-harvard) [reply pyproject.toml:24]: Okay, will have to make dcicutils be >=3.8 too which requires downgrading flake8, etc; I can see how that goes.

Comment (alexander-veit) [reply pyproject.toml:24]: Oh, I see. You can leave 3.8.1 then. It's just us using it. Should be fine.

File: .github/workflows/main.yml (hunk 1 above)
Comment (dmichaels-harvard) [reply .github/workflows/main.yml]: Yeah I should put that back, with some repos careful to include minimal set due to time, but this is pretty quick.

File: pyproject.toml (hunk 3 above)
Comment (dmichaels-harvard) [reply pyproject.toml]: Removed; was artifact of some troubleshooting of some problem; not needed there though.

**Review by alexander-veit — APPROVED** [review]

**Pull request merged**